int32 play_number
int32 amount

